server/submission/definitions.js

```diff
@@ -167,4 +167,121 @@ const emptyManuscript = {
+            email: Joi.string()
+              .email()
+              .required(),
+            institution: Joi.string().required(),
+          })
+          .required(),
+        hasCorrespondent: Joi.boolean().required(),
+        correspondent: Joi.when('hasCorrespondent', {
+          is: true,
+          then: Joi.object()
+            .keys({
+              firstName: Joi.string().required(),
+              lastName: Joi.string().required(),
+              email: Joi.string().required(),
+              institution: Joi.string().required(),
+            })
+            .required(),
+        }),
+        stage: Joi.string()
+          .valid(possibleStages)
+          .required(),
+        discussedPreviously: Joi.boolean().required(),
+        discussion: Joi.when('discussedPreviously', {
+          is: true,
+          then: Joi.string().required(),
+        }),
+        consideredPreviously: Joi.boolean().required(),
+        previousArticle: Joi.when('consideredPreviously', {
+          is: true,
+          then: Joi.string().required(),
+        }),
+        cosubmission: Joi.boolean().required(),
+        cosubmissionTitle: Joi.string().allow(''),
+        cosubmissionId: Joi.string().allow(''),
+      })
+      .required(),
+    suggestedSeniorEditors: Joi.array()
+      .items(Joi.string().required())
+      .required(),
+    opposedSeniorEditors: Joi.array().items(
+      Joi.object().keys({
+        name: Joi.string().required(),
+        reason: Joi.string().required(),
+      }),
+    ),
+    suggestedReviewingEditors: Joi.array()
+      .items(Joi.string().required())
+      .required(),
+    opposedReviewingEditors: Joi.array().items(
+      Joi.object().keys({
+        name: Joi.string().required(),
+        reason: Joi.string().required(),
+      }),
+    ),
+    suggestedReviewers: Joi.array().items(
+      Joi.object()
+        .keys({
+          name: Joi.string().required(),
+          email: Joi.string()
+            .email()
+            .required(),
+        })
+        .required(),
+    ),
+    opposedReviewers: Joi.array().items(
+      Joi.object().keys({
+        name: Joi.string().required(),
+        email: Joi.string()
+          .email()
+          .required(),
+        reason: Joi.string().required(),
+      }),
+    ),
+    noConflictOfInterest: Joi.boolean().required(),
+  })
+  .required()
+
+const cosubmissionSchema = Joi.alternatives().try(
+  Joi.object().keys({
+    cosubmission: Joi.boolean().required(),
+    cosubmissionTitle: Joi.string().when('cosubmission', {
+      is: true,
+      then: Joi.required(),
+      otherwise: Joi.allow(''),
+    }),
+    cosubmissionId: Joi.string().allow(''),
+  }),
+  Joi.object().keys({
+    cosubmission: Joi.boolean().required(),
+    cosubmissionTitle: Joi.string().allow(''),
+    cosubmissionId: Joi.string().when('a', {
+      is: true,
+      then: Joi.required(),
+      otherwise: Joi.string().allow(''),
+    }),
+  }),
+)
```

Comment:
Surely we should be reusing the yup validation schema instead of duplicating it all in joi?

Reply:
I assume we still want to use Joi since this is backend stuff? I'll look into how we can easily convert between the two.

Reply:
Why do you assume that?

Reply:
For a couple of reasons: PubSweet already uses Joi for validation on the backend. Keeping the number of packages we use across the entire project to a minimum sounds like a good thing. Second, coming from PubSweet to elife-xpub could be more confusing to a new developer if we switch from Joi to Yup all of a sudden. Third, they are not entirely identical, notice that this one has stage required, while the frontend one doesn't.

Also, something similar was previously discussed: https://gitlab.coko.foundation/xpub/xpub-elife/merge_requests/11

Your own words here :)
> I think reusing validation schemas between server and client is a non-goal so I'd be happy with using different libs on the server and the client.

But I do think we should store the schema in one place only and maybe slightly change it for frontend/backend stuff. I'm open to discussions here. I would be OK with using yup on both frontend and backend if everyone agrees that the arguments I mentioned above are not strong enough.

Reply:
Don't quote me, I don't know anything!

I think the difference is that in PubSweet the Joi validation schemas are intended to determine whether the entity as a whole is in a valid state. Here we're only interested in whether the user submitted data is valid which is what the yup schema tells us. Extra fields which are not supplied by the user (like stage) don't need to be validated. 

However the current schema doesn't account for all possibilities. For instance it doesn't validate that `submissionMeta.author` is required because in the context of the form that's not an error that can occur. I think it should be possible to augment the schema to validate these cases but it's worth checking to make sure.

Reply:
As per our discussion, this was moved to https://github.com/elifesciences/elife-xpub/issues/206